Added the diagonal sum code

Soumyosish opened the pull request:
<!-- 
To check✅ a checlist, fill x or X in the [ ]
Like, - [x] or - [X]
 -->
I have added a problem to solve the diagonal sum of main and anti diagonals
## Issue number if any:-
Fixes #31 

## ✅ **Contribution Checklist (make sure all are checked)**

- [x] Code file added (`.c`, `.java`)
- [x] Markdown file added with:
  - [x] Problem statement  
  - [x] Example input/output  
  - [x] Generic logic  
  - [x] Time & space complexity  

---

## 📝 **Type of Addition (check any one)**

- [x] DSA problem addition  
- [ ] Fix mistakes in existing codebase
- [ ] Add better solutions  
- [ ] Make docs better  

---

## 📊 **Last check** (make sure all are checked

- [x] **Have you checked that this problem isn't already in the codebase?🔍**
- [x] Did you cross-check that your code/explanation is right?

## 🏢 **Your Organization?**
- [x] GsSOC'25
Please @shivamm-verma check and merge the PR also assign the level of contribution in the PR an GSSOC 25 label. Let me know if any further changes are required.

shivamm-verma reviewed (changes requested):
- [ ] Please remove .vscode folder
- [ ] Make the folder name 9_

shivamm-verma commented:
It is compulsory to add the markdown.md
Checkout this [Example](https://github.com/shivamm-verma/Daily-DSA/blob/main/1_Check%20if%20a%20String%20is%20a%20Palindrome/Markdown.md)

Soumyosish commented:
I have removed the .vscode folder and renamed the folder to 9 also removed .class and .exe file and added markdown.md .
@shivamm-verma check once again

shivamm-verma commented:
Can you also remove unnecessary comments in the coding files? they look chatgptied
Simple and clean code files are required.

Soumyosish commented:
I had kept the comments for better understanding of the user. Okay I have removed the comments and made it clean. Check @shivamm-verma once again.

shivamm-verma commented:
> I had kept the comments for better understanding of the user. Okay I have removed the comments and made it clean. Check @shivamm-verma once again.

There were a lot of unnecessary comments in it. It's okay under `Better-solution`, someone can add them again, in future.

shivamm-verma commented:
Make sure to give right and **related** commit messages. It's a convention!

shivamm-verma commented:
Congrats on getting your first PR merged! 🥳

Soumyosish commented:
Please @shivamm-verma change the GsSoc label to either gssoc or GSSoC otherwise the PR will not be counted in the leaderboard
